changelog.md

```diff
@@ -30,6 +31,7 @@
 * Fixed bug where resources were not cleaned up if connect was called twice
 * Fixed bug where serial data was written after disconnection
 * Fixed bug where unexpected disconnection was not detected
+* Fixed bug where promises used in packet processing allowed for context/process switch.
```

Comment:
I mis-spoke earlier; javascript doesn't actually allow for context switching, and can't prevent process switching, but the change fixes possible mis-ordering of responses since the order in which promises are resolved is not guaranteed